App/views.py

```diff
@@ -0,0 +1,76 @@
+import json
+import requests
+import urllib.parse
+
+from django.http import HttpResponse
+from django.shortcuts import render
+
+
+def user_revisions_analysis(request):
+    """
+    Display analysis of last 5 edits made by user.
+    """
+    if request.method=='POST':
+        # Here, we get the username
+        username = request.POST['username']
+        # if username is submitted blank 
+        if not username:
+            return render(request, "App/task4.html", {"error": "Please enter a username"})
+        # if username is not blank
+        details = list()
+        parameters = {'action':'query',
+                      'format':'json',
+                      'list':'usercontribs',
+                      'uclimit':'5',
+                      'ucuser':username}
+        url_base = 'https://en.wikipedia.org/w/api.php?'
+        # Recieved data in json-format
+        response = requests.get(url_base, params = parameters)
+        if response.status_code == 200:
+            try:
+                # Edit data extracted
+                edits_data = json.loads(response.text)
+                if 'error' in edits_data:
+                    return(render(request,
```

Comment:
This works just fine, but you could also just do something like

    if 'error' in edits_data:
        raise Error(edits_data['error']['info'])

which is arguably easier to read (all the error rendering happens in one place). As your code becomes more complex, you might want to separate rendering form the main control flow (use [mode-view-controller](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller) or something similar) and exceptions are one way to handle it.

Reply:
Thanks for the suggestion. I have corrected code to handle all error rendering in except branch.